### PR TITLE
Check the cipher flags to see if the cipher supports aead

### PR DESCRIPTION
**Pull request by danielwestendorf**
Supports checking if a cipher supports aead, with an `authenticated?` helper method.

### CODE REVIEW

**Comment by danielwestendorf** [comment]
Need is explained [here](https://github.com/crystal-lang/crystal/issues/4593#issuecomment-414342989).

**Comment by danielwestendorf** [comment]
The `darwin` CI suite seems to be failing because `aes-128-gcm` isn't an available cipher, however, when run locally on my macOS system (with Crystal installed via homebrew) it is available and the spec passes.

**Comment by ysbaddaden** [comment]
CI seems to run on macOS sierra, which probably has an old OpenSSL version (maybe 0.9.8zh).

Do you know which OpenSSL versions support `aes-128-gcm`?

**Comment by danielwestendorf** [comment]
@ysbaddaden I think you're on to something there. I'm trying different macOS/OSX versions to see what the oldest version this change would require is, as I can't seem to find any definitive information about when `aes-128-gcm` was added to OpenSSL.

File: .circleci/config.yml [hunk 1]
```diff
@@ -447,4 +447,3 @@ workflows:
             - dist_darwin
             - dist_docker
             - dist_docs
-
```
Comment (asterite) [inline .circleci/config.yml]: This line should be kept.

Comment (danielwestendorf) [reply .circleci/config.yml]: Whoops!

File: spec/std/openssl/cipher_spec.cr [hunk 2]
```diff
@@ -44,4 +44,18 @@ describe OpenSSL::Cipher do
     s3.to_s.should eq(data)
     s3.to_slice.should eq(s4.to_slice)
   end
+
+  it "authenticated?" do
+    begin
+      cipher = OpenSSL::Cipher.new("aes-128-gcm")
+      cipher.authenticated?.should eq(true)
+    rescue ArgumentError
```
Comment (straight-shoota) [inline spec/std/openssl/cipher_spec.cr]: Maybe add a check that its actually an expected ArgumentError by verifying the error message. It should be re-raised if it doesn't match.

Comment (danielwestendorf) [reply spec/std/openssl/cipher_spec.cr]: Could definitely do that, however if the `ArgumentError` is originating from anything other than the cipher not being available, it'd raise on the next invocation (line 58/59).

**Comment by bcardiff** [comment]
I'm late to the party. The changes are fine. At most I would've try to only accept the exception as a success if the error message is certain, or wrap the whole block of the `it` in a feature check `if`.

**Pull request merged**